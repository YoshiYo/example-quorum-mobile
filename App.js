import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native';
import { Provider } from 'react-redux';
import store from './src/_store/index';
import Welcome from './src/_containers/welcome/welcome';
import Header from './src/components/header/header';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#fff',
  },
});

class App extends Component {
  render() {
    return (
      // Adding a store to give data to the app
      // It comes from reducers
      // Provider function envelopp the Global App
      <Provider store={store}>
        <View style={styles.container}>
          <Header />
          <Welcome />
        </View>
      </Provider>
    );
  }
}

export default App;
