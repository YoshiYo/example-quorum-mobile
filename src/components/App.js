import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native';
import I18n from 'ex-react-native-i18n';
import PropTypes from 'prop-types';
import Welcome from '../_containers/welcome/welcome';
import Header from './header/header';
import languages from '../../static/lang/index';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#fff',
  },
});

I18n.translations = {
  ...languages,
};

class App extends Component {
  async componentWillMount() {
    const { setupI18n } = this.props;
    I18n.initAsync();
    I18n.fallbacks = true;
    setupI18n(I18n);
  }

  render() {
    const { langReady } = this.props;

    if (!langReady) {
      return (
        <View style={styles.container} />
      );
    }
    return (
      // Adding a store to give data to the app
      // It comes from reducers
      // Provider function envelopp the Global App
      <View style={styles.container}>
        <Header />
        <Welcome />
      </View>
    );
  }
}

App.propTypes = {
  langReady: PropTypes.bool.isRequired,
};

export default App;
