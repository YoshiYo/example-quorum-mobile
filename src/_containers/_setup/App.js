import { connect } from 'react-redux';
import App from '../../components/App';
import * as actions from '../../_actions/SetupActions';

const mapStateProps = (state) => {
  const stateForStore = {
    I18n: state.setup.I18n,
    langReady: state.setup.langReady,
  };
  return stateForStore;
};

const mapDispatchProps = (dispatch) => {
  const functionsForStore = {
    setupI18n: (I18n) => {
      dispatch(actions.setupI18n(I18n));
    },
  };
  return functionsForStore;
};

export default connect(mapStateProps, mapDispatchProps)(App);