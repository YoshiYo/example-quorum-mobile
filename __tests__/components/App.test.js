import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import App from '../../App';
import Header from '../../src/components/header/header';

Enzyme.configure({ adapter: new Adapter() });

// import {createSerializer} from 'enzyme-to-json';
// expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

describe('Homepage rendered', () => {
  it('Render the homepage and watch changes', () => {
    const wrapper = shallow(
      <App />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('Render an Header', () => {
    const wrapper = shallow(
      <App />,
    );
    expect(wrapper.find(Header));
  });
});
