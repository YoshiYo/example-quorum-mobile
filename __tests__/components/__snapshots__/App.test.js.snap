// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage rendered Render the homepage and watch changes 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Component
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Header />
    <Connect(Welcome) />
  </Component>
</Provider>
`;
